pages/glossary/default-element-language.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Default Element Language
+key: default-element-language
+unambiguous: true
+objective: true
+input_aspects:
+  - DOM tree
+  - Accessibility tree
+  - CSS Styling
+  - Language
+---
+
+The _default language of an element_ E is the _most common language_ used in the texts of its _candidate elements_. If there are no _words_ in the candidate elements, the default language is undefined.
+
+The _most common language_ is determined by counting the number of _words_ in the candidate elements that are part of any of the languages in the [language subtag registry][]. The same word can be part of multiple languages. If the number of words is the same, the default language is undefined.
+
+The _candidates elements_ are all the elements for which all the following are true:
+
+- the element is an [inclusive descendant][] of E in the [flat tree][]; and
+- the element has no [inclusive ancestor][] in the [flat tree][] which is a descendant of E and has a non-empty (`""`) `lang` attribute (that is, no `lang` attribute "between" them).
```

Comment:
Should it just be non-empty or should it also be a valid language tag? I'm leaning towards the latter

Reply:
Current definition is using "non-empty": https://act-rules.github.io/glossary/#default-page-language
I didn't want to change that (I haven't put much thoughts in it…)

I guess this is about how ATs are handling invalid language tags, whether they default to the previous language or do something else.
I'm summoning @WilcoFiers on that one. He wrote #1334 and hopefully can remember why he chose "non-empty" rather than "valid".

Reply:
Digging into this, an invalid language tag should be handled as its own language and carried as is (essentially assuming that something down the line (e.g. a CSS selector) knows what to do about it), so I think that "non-empty" is the correct stuff to use.

See example at https://html.spec.whatwg.org/multipage/dom.html#attr-lang and the paragraph before it.